pkg/nvlib/device/device.go

```diff
@@ -140,6 +142,29 @@ func (d *device) GetBrandAsString() (string, error) {
 	return "", fmt.Errorf("error interpreting device brand as string: %v", brand)
 }
 
+// GetBusID returns the string representation of the bus ID.
+func (d *device) GetBusID() (string, error) {
+	info, ret := d.GetPciInfo()
+	if ret != nvml.SUCCESS {
+		return "", fmt.Errorf("error getting PCI info: %w", ret)
+	}
+
+	var bytes []byte
+	for _, b := range info.BusId {
+		if byte(b) == '\x00' {
+			break
+		}
+		bytes = append(bytes, byte(b))
+	}
+	id := strings.ToLower(string(bytes))
+
+	if id != "0000" {
+		id = strings.TrimPrefix(id, "0000")
+	}
```

Comment:
If I recall correctly, some systems had 8 leading numbers and some 4. I guess this is handling the case when there are 8 leading numbers but the first 4 are 0000. Is that correct?

Reply:
Yes. That's correct.